march_data_collector/src/march_data_collector/cp_calculator.py

```diff
@@ -64,15 +65,11 @@ def center_of_mass_marker(self, updated_center_of_mass):
         if self._delta_t == 0:
             return
 
-        self.vx = (updated_center_of_mass.pose.position.x - self._center_of_mass_marker.pose.position.x) / self._delta_t
-        self.vy = (updated_center_of_mass.pose.position.y - self._center_of_mass_marker.pose.position.y) / self._delta_t
+        self.vx = (updated_center_of_mass.pose.position.x - self._center_of_mass.x) / self._delta_t
+        self.vy = (updated_center_of_mass.pose.position.y - self._center_of_mass.y) / self._delta_t
 
-        self.x = self._center_of_mass_marker.pose.position.x
-        self.y = self._center_of_mass_marker.pose.position.y
-        self.z = self._center_of_mass_marker.pose.position.z
-
-        self._center_of_mass_marker = copy.deepcopy(updated_center_of_mass)
-        self._prev_t = updated_center_of_mass.header.stamp
+        self._center_of_mass = updated_center_of_mass.pose.position
```

Comment:
```suggestion
        self._center_of_mass = copy.deepcopy(updated_center_of_mass.pose.position)
```
This deepcopy was here with a reason :innocent: (don't forget to add the import)